### PR TITLE
EN-19151: Support query rewrite on count(column).

### PR DESCRIPTION
**Pull request by znep**
A count on a specific column is the number of non-null values, which can be
added together with a simple sum just like count(*).

### CODE REVIEW

**Comment by socrata-pull-request-updater** [comment]
This PR should be reviewed by: @socrata/dataplatform

File: query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala [hunk 1]
```diff
@@ -86,6 +86,14 @@ class QueryRewriter(analyzer: SoQLAnalyzer[SoQLType]) {
         })
   }
 
+  /** Find the first column index that exactly matches the given function call / params / etc. */
+  def findFunctionOnColumn(rollupColIdx: Map[Expr, Int], queryFc: FunctionCall): Option[Int] = {
+    rollupColIdx.find {
```
[the remark this answers is not in the document]
Comment (znep) [reply query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala]: D'oh!  Can't see the trees for the forest.

Fixed.

**Comment by znep** [comment]
Bumped soql-reference as well.

**Comment by chitang** [comment]
:shipit:

**Pull request merged**